### ISSUE TITLE
`chain_sim()` fails with a negative binomial and `mu = 0` and `size = 0`

### ISSUE THREAD

**Issue by joshwlambert** [epiforecasts/bpmodels#149]
This is an unlikely, and epidemiologically nonsensical, use case, but when running `bpmodels::chain_sim()` with $R = 0$ and $k = 0$ then the function fails with an unfriendly error message.

``` r
bpmodels::chain_sim(n = 10, offspring = "nbinom", stat = "size", mu = 0, size = 0)
#> Warning in get(roffspring_name)(n = sum(n_offspring[sim]), ...): NAs produced
#> Error in if (any(next_gen%%1 > 0)) {: missing value where TRUE/FALSE needed
```

<sup>Created on 2024-08-22 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

It would be good to catch this with an input check.

**Comment by jamesmbaazam** [comment]
Thanks for reporting, Josh. I will add a quick check here for negative binomial parameters as we discussed in person but as noted in the README, we have transitioned all features of bpmodels to the [epichains](https://epiverse-trace.github.io/epichains/) package, which we strongly recommend that users use moving forward.

For example, running the same thing with `{epichains}` yields this error which could potentially be improved in not clear.
``` r
library(epichains)
epichains::simulate_chain_stats(
  n_chains = 10,
  offspring_dist = rnbinom,
  statistic = "size",
  mu = 0,
  size = 0
  )
#> Warning in (function (n, size, prob, mu) : NAs produced
#> Error in .sample_possible_offspring(offspring_func = offspring_dist, offspring_func_pars = pars, : Offspring distribution must return integers
```

<sup>Created on 2024-08-22 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by joshwlambert** [comment]
Thanks, the fix in #150 looks good to me.
